staging/src/kubevirt.io/api/instancetype/v1beta1/types.go

```diff
@@ -74,6 +74,20 @@ type VirtualMachineClusterInstancetypeList struct {
 //
 // CPU and Memory are required attributes with both requiring that their Guest attribute is defined, ensuring a number of vCPUs and amount of RAM is always provided by each instancetype.
 type VirtualMachineInstancetypeSpec struct {
+	// NodeSelector is a selector which must be true for the vmi to fit on a node.
+	// Selector which must match a node's labels for the vmi to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	//
+	// NodeSelector is the name of the custom node selector for the instancetype.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, the VMI will be dispatched by specified scheduler.
+	// If not specified, the VMI will be dispatched by default scheduler.
+	//
+	// SchedulerName is the name of the custom K8s scheduler for the instancetype.
+	// +optional
+	SchedulerName string `json:"schedulerName,omitempty"`
```

Comment:
Can you move this to a string pointer as it's optional and easier to tell that a user has actually provided something and not just an empty string.

Reply:
> easier to tell that a user has actually provided something and not just an empty string.

What would be the point here? Empty `schedulerName` is valid option in Kubernetes world which translates to `default-scheduler`.

Reply:
Hm, after some thought this could be indeed useful for conflict resolution.

Reply:
Thanks, we (the KubeVirt project) likely need to write down some recommendations such as using pointers for optional attributes as it isn't universally followed in the API and we often rely on the empty values of types.

Reply:
Hm, after another thought, this wouldn't work. Because you have no ways of knowing if `vmiSpec.SchedulerName` set to `""` because user set it or because it was default.

Because ideally, we could do:
```
	if instancetypeSpec.SchedulerName == nil {
		return nil
	}

	if vmiSpec.SchedulerName != nil { // `vmiSpec.SchedulerName` should be also `*string`.
		return Conflicts{field.Child("schedulerName")}
	}
```
but now it should be:
```
	if instancetypeSpec.SchedulerName == nil {
		return nil
	}

	if vmiSpec.SchedulerName != "" {
		return Conflicts{field.Child("schedulerName")}
	}
```
and that's looks good expect the case when `vmiSpec.SchedulerName == ""` and `*instancetypeSpec.SchedulerName == ""`. Should it be conflict in that situation? Because user explicitly provided empty value in `instancetypeSpec`. But we have no way of knowing if `vmiSpec.SchedulerName` is indeed empty because of the user value or default because it's default value for `string`.

So I wouldn't change the type of this field. And argue that empty value `""` is empty because user didn't set it. And if user wants to set it then `default-scheduler` value should be used.

Reply:
Apologies again for the delay getting back to this.

Yeah understood, not moving both to string pointers and ignoring empty values would still lead to `default-scheduler` getting picked up eventually so that's okay.